src/detours.cpp

```diff
@@ -1959,12 +2013,287 @@ LONG WINAPI DetourUpdateThread(_In_ HANDLE hThread)
     }
 
     t->hThread = hThread;
+    t->fCloseThreadHandleOnDestroyDetourThreadObject = fCloseThreadHandleOnDestroyDetourThreadObject;
     t->pNext = s_pPendingThreads;
     s_pPendingThreads = t;
 
     return NO_ERROR;
 }
 
+// Code modified from https://github.com/apriorit/mhook/blob/master/mhook-lib/mhook.c
+//=========================================================================
+// ntdll definitions
+
+typedef LONG NTSTATUS;
+
+typedef LONG KPRIORITY;
+
+typedef enum _SYSTEM_INFORMATION_CLASS
+{
+    SystemBasicInformation = 0,
+    SystemPerformanceInformation = 2,
+    SystemTimeOfDayInformation = 3,
+    SystemProcessInformation = 5,
```

Comment:
Only declare what we need?

Reply:
These are also declared in `"C:\Program Files (x86)\Windows Kits\10\Include\10.0.19041.0\um\winternl.h"`, so you should not worry about leaking too much information in the Windows kernel to users. These are kept just to be consistent with the original source of the code, of course, I can change it if you feel unnecessary.

Reply:
the latest commit have remove the code from mhook.